tools/oeedger8r/Emitter.ml

```diff
@@ -1427,16 +1432,26 @@ let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
     ; "    const char* path,"
     ; "    oe_enclave_type_t type,"
     ; "    uint32_t flags,"
+    ; "#ifdef OE_CONTEXT_SWITCHLESS_EXPERIMENTAL_FEATURE"
+    ; "    const oe_enclave_config_t* configs,"
+    ; "    uint32_t config_count,"
+    ; "#else"
     ; "    const void* config,"
     ; "    uint32_t config_size,"
+    ; "#endif"
     ; "    oe_enclave_t** enclave)"
     ; "{"
     ; "    return oe_create_enclave("
     ; "               path,"
     ; "               type,"
     ; "               flags,"
+    ; "#ifdef OE_CONTEXT_SWITCHLESS_EXPERIMENTAL_FEATURE"
```

Comment:
We need to add the release barrier in emitted code when the arg struct's result field is set.

Reply:
There are two `result` fields in two separate structures. The emitted code sets the `result` field in `host_<ocall_function_name>_args_t`. But that's not the sync variable between the host thread and the enclave thread.  

I did add the write barrier in `oe_handle_call_host_function` before setting `oe_call_host_function_args_t.result`, the sync variable we do care about.